internal/elasticsearch/security/role.go

```diff
@@ -418,6 +561,29 @@ func flattenIndicesData(indices *[]models.IndexPerms) []interface{} {
 	}
 	return make([]interface{}, 0)
 }
+func flattenRemoteIndicesData(remoteIndices *[]models.RemoteIndexPerms) []interface{} {
```

Comment:
Why `remoteIndices *[]models.RemoteIndexPerms` here? It looks like the only usage takes the address of a plain slice. 

We can remove the if block and drop a level of nesting without the pointer typed variable here.

Reply:
Ok, it looks like you're just mimicking the current code. IMO we should change this to just take a direct slice reference (which can be nil anyway). It also means we don't have the `remoteIndexes`/`remoteIndices` naming above which confused me for longer than it should have.